spark-sql-application/src/main/scala/org/opensearch/sql/SQLJob.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql
+
+import org.apache.spark.sql.{DataFrame, SparkSession, Row}
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.functions._
+
+object SQLJob {
+  def main(args: Array[String]) {
+    // Get the SQL query and Opensearch Config from the command line arguments
+    val query = args(0)
+    val index = args(1)
+    val host = args(2)
+    val port = args(3)
+    val scheme = args(4)
+    val auth = args(5)
+    val region = args(6)
+
+    // Create a SparkSession
+    val spark = SparkSession.builder().appName("SQLJob").getOrCreate()
+
+    try {
+      // Execute SQL query
+      val result: DataFrame = spark.sql(query)
+
+      val resultJson = getJson(result)
+      val schemaJson = getJson(getSchema(spark, result))
+      val data = getList(spark, resultJson, "result").join(getList(spark, schemaJson, "schema")).withColumn("stepId", lit(sys.env.getOrElse("EMR_STEP_ID", "")))
```

Comment:
q: why the join and aggregation required? does the sample code below to the same thing?

```
import org.json4s.JsonDSL._
import org.json4s.jackson.JsonMethods._

val df = spark.sql("select 1")
val jsonObject = ("result" -> df.toJSON.collect.toList)
val jsonString = compact(render(jsonObject))

## output
{"result":["{\"1\":1}"]}
```

Reply:
Thanks, this works too. Updated code accordingly.